app/main.py

```diff
@@ -1,6 +1,31 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_store = {}
+
+    @wraps(func)
+    def wrapper(*args):
+        key = tuple(args)
+        if key in cache_store:
+            print("Getting from cache")
+            return cache_store[key]
+
+        print("Calculating new result")
+        result = func(*args)
+        cache_store[key] = result
+        return result
+
+    return wrapper
+
+
+@cache
+def long_time_func(a: int, b: int, c: int) -> int:
+    return (a ** b ** c) % (a * c)
+
+
+@cache
+def long_time_func_2(n_tuple: tuple, power: int) -> int:
+    return [number ** power for number in n_tuple]
```

Comment:
The function `long_time_func_2` is expected to return an `int` as per its type hint, but it currently returns a list. You should either change the return type to `List[int]` or modify the function to return an integer.